api/grpc_disruption_test.go

```diff
@@ -0,0 +1,345 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package api_test
+
+import (
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("GRPCDisruption Validation", func() {
+	It("Error and override cannot both be defined", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "{}",
+					QueryPercent:     100,
+				},
+			},
+		}
+
+		err := spec.Validate()
+		Expect(err.Error()).To(Equal("the gRPC disruption has ErrorToReturn and OverrideToReturn specified for endpoint /chaos_dogfood.ChaosDogfood/order, but it can only have one"))
+	})
+
+	It("Error and override cannot both be defined", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "",
+					OverrideToReturn: "",
+					QueryPercent:     100,
+				},
+			},
+		}
+
+		err := spec.Validate()
+		Expect(err.Error()).To(Equal("the gRPC disruption must have either ErrorToReturn or OverrideToReturn specified for endpoint /chaos_dogfood.ChaosDogfood/order"))
+	})
+
+	It("Query percent of 100 does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     100,
+				},
+			},
+		}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent of 99 does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     99,
+				},
+			},
+		}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent of 100 on two endpoints does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     100,
+				},
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/getCatalog",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     100,
+				},
+			}}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent which sums to 100 on two endpoints does validate", func() {
```

Comment:
```suggestion
	It("Query percents which sum to 100 on one endpoint does validate", func() {
```

There's inconsistency in these test names. Sometimes "two endpoints" means two distinct endpoints, but other times it means two alterations for a single endpoint